### PR TITLE
Fix staticcheck Linter Errors

### PR DESCRIPTION
**Pull request by relyt29**
This change goes through and fixes bugs discovered by systematic running of the golang `staticcheck` tool.

## Testing and Verifying

This PR will be tested by our existing test suite, and / or is an addendum to our existing test suite. It is not meant to change the functionality of the chain or introduce new features.

**REVIEWER NOTE:** I commented out the unused functions `scheduleUpgrades` in `app/upgrades.go` and `removeOldKVStores` in `x/emissions/migrations/v2/migrate.go` - I don't think either are necessary or correct for v0.3.0 release but I kept them commented out in case they're needed for reference. If that was the wrong thing to do, please lets figure out a better solution.

## Documentation and Release Note

No documentation implications.

### CODE REVIEW

File: x/emissions/migrations/v2/migrate.go [hunk 1]
```diff
@@ -292,8 +292,9 @@ func restoreAllRecordCommits(store storetypes.KVStore, cdc codec.BinaryCodec, co
...
-	store.Delete(types.ChurnableTopicsKey)
-	store.Delete(types.TopicLastWorkerPayloadKey)
-	store.Delete(types.TopicLastReputerPayloadKey)
-}
+// staticcheck is saying this is never used, can we delete it??
+// func removeOldKVStores(store storetypes.KVStore) {
+// 	store.Delete(types.ChurnableTopicsKey)
+// 	store.Delete(types.TopicLastWorkerPayloadKey)
+// 	store.Delete(types.TopicLastReputerPayloadKey)
+// }
```
Comment (kpeluso) [inline x/emissions/migrations/v2/migrate.go:300]: @fernandofcampos @guilherme-brandao @RedBird96  ?? probably should be called above

**Review by kpeluso — COMMENTED** [review]
mostly good apart from merge conf + 1 comment

File: x/emissions/migrations/v2/migrate.go (hunk 1 above)
Comment (guilherme-brandao) [reply x/emissions/migrations/v2/migrate.go:300]: this is an old version I guess

Comment (guilherme-brandao) [reply x/emissions/migrations/v2/migrate.go]: fixed it

**Review by kpeluso — APPROVED** [review]

**Pull request merged**